src/algorithms/crypto/randomness.go

```diff
@@ -33,8 +33,10 @@ func (tag DomainSeparationTag) DeriveRandWithIndex(s Serialization, index int) R
 func _deriveRandInternal(tag DomainSeparationTag, s Serialization, index int) Randomness {
 	buffer := []byte{}
 	buffer = append(buffer, util.IntToBytesLittleEndian(int(tag))...)
-	buffer = append(buffer, util.IntToBytesLittleEndian(int(index))...)
 	buffer = append(buffer, Bytes(s)...)
+	if index != -1 {
+		buffer = append(buffer, util.IntToBytesLittleEndian(int(index))...)
+	}
```

Comment:
This change would break the unambiguity proof (https://github.com/filecoin-project/specs/pull/679#discussion_r352290621) -- the serialization may be variable-length.

Reply:
ok. I'll flip order again. But couldn't the index? We are talking about chain epochs here which could grow v large.

Reply:
No, the index is 64 bits and chain epochs should never exceed that. (Also it's not just a matter of exchanging the order -- we have to always include the index unconditionally in the serialization, whether or not it's -1.)